java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java

```diff
@@ -35,8 +44,73 @@
 public class TaskManagerClient {
 
     private RpcClient rpcClient;
+    private CuratorFramework zkClient;
     private TaskManagerInterface taskManagerInterface;
+    private static final Log logger = LogFactory.getLog(TaskManagerClient.class);
+
+    public TaskManagerClient(String zkCluster, String zkPath) throws Exception {
+        if (zkCluster == null || zkCluster.length() == 0 || zkPath == null || zkPath.length() == 0) {
+            logger.info("Zookeeper address is wrong, please check the configuration");
+        }
+        String masterZnode = zkPath + "/taskmanager" + "/leader";
+
+        zkClient = CuratorFrameworkFactory.builder()
+                .connectString(zkCluster)
+                .sessionTimeoutMs(10000)
+                .retryPolicy(new ExponentialBackoffRetry(1000, 10))
+                .build();
+        zkClient.start();
+        Stat stat = zkClient.checkExists().forPath(masterZnode);
+        if (stat != null) {  // The original master exists and is directly connected to it.
+            String endpoint = new String(zkClient.getData().forPath(masterZnode));
+            watch(zkClient, masterZnode);
+            this.rpcConnect(endpoint);
+        } else {
+            throw new Exception("TaskManager has not started yet, connection failed");
+        }
+    }
+    /**
+     * Create a data change listener event for this node.
+     *
+     * @param curator the zkClient.
+     * @param path the path of zkNode.
+     */
+    public static void watch(CuratorFramework curator, String path) throws Exception {
+        final NodeCache nodeCache = new NodeCache(curator, path, false);
+        nodeCache.getListenable().addListener(new NodeCacheListener() {
+            @Override
+            public void nodeChanged() throws Exception {
+                System.out.println("The content of the node was changed or deleted, please try to reconnect");
+                System.exit(0);
```

Comment:
Do not exit when `nodeChanged`. Just reset the rpc client with the new TaskManager server.